<launch>
    <arg name="eye_in_hand" default="true"/>
    <arg name="tracking_base_frame" value="tracking_origin"/>
    <arg name="tracking_marker_frame" value="tracking_marker"/>
    <arg name="robot_base_frame" value="panda_link0"/>
    <arg name="robot_effector_frame" value="panda_link8"/>

    <!-- start the robot -->
    <include file="$(find-pkg-share easy_handeye2_demo)/launch/run_move_group.launch.py"/>

    <!-- start the tracking -->
    <node pkg="easy_handeye2_demo" exec="tracking_simulator" name="tracking_simulator">
        <!-- simulator arguments -->
        <param name="frequency" value="10"/>
        <param name="translation_noise_stdev" value="0.001"/>
        <param name="rotation_noise_stdev" value="0.0001"/>
        <param name="base_to_tracking" value="1 0 0.5 0 0 0 1"/>
        <param name="hand_to_tracking" value="0.12 0.21 0.137 0 0 0 1"/>

        <!-- calibration arguments -->
        <param name="eye_in_hand" value="$(var eye_in_hand)"/>
        <param name="tracking_base_frame" value="$(var tracking_base_frame)"/>
        <param name="tracking_marker_frame" value="$(var tracking_marker_frame)"/>
        <param name="robot_base_frame" value="$(var robot_base_frame)"/>
        <param name="robot_effector_frame" value="$(var robot_effector_frame)"/>
    </node>
</launch>
